accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicMonotonicTestForAlternateRanges.java

```diff
@@ -0,0 +1,121 @@
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import uk.ac.ebi.ampt2d.commons.accession.autoconfigure.MonotonicGeneratorProperties;
+import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionCouldNotBeGeneratedException;
+import uk.ac.ebi.ampt2d.commons.accession.generators.DecoratedAccessionGenerator;
+import uk.ac.ebi.ampt2d.commons.accession.generators.monotonic.MonotonicAccessionGenerator;
+import uk.ac.ebi.ampt2d.commons.accession.hashing.SHA1HashingFunction;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.ContiguousIdBlockService;
+import uk.ac.ebi.ampt2d.commons.accession.service.BasicMonotonicAccessioningService;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestMonotonicDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.service.TestMonotonicDatabaseService;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestMonotonicDatabaseServiceTestConfiguration.class})
+@TestPropertySource("classpath:application-test1.properties")
+public class BasicMonotonicTestForAlternateRanges {
+
+    @Autowired
+    private TestMonotonicDatabaseService databaseService;
+
+    @Autowired
+    private ContiguousIdBlockService contiguousIdBlockService;
+
+    @Autowired
+    @Qualifier("getMonotonicGeneratorPropertiesForEva")
+    private MonotonicGeneratorProperties monotonicGeneratorProperties;
+
+    @Test
+    public void testAlternateRangesWithPrefixes() throws AccessionCouldNotBeGeneratedException {
+        Map<String, TestModel> objects = new LinkedHashMap<>();
+        objects.put("hash1", TestModel.of("service-test-1"));
+        objects.put("hash2", TestModel.of("service-test-2"));
+        objects.put("hash3", TestModel.of("service-test-3"));
+        objects.put("hash4", TestModel.of("service-test-4"));
+        objects.put("hash5", TestModel.of("service-test-5"));
+        objects.put("hash6", TestModel.of("service-test-6"));
+        DecoratedAccessionGenerator<TestModel, Long> generator = DecoratedAccessionGenerator
+                .buildPrefixSuffixAccessionGenerator(getGenerator(), "RS", null, Long::parseLong);
+        List<AccessionWrapper<TestModel, String, String>> generated = generator.generateAccessions(objects);
+        assertEquals(6, generated.size());
+        assertEquals("RS1", generated.get(0).getAccession());
+        assertEquals("RS2", generated.get(1).getAccession());
+        assertEquals("RS11", generated.get(5).getAccession());
```

Comment:
Even if this looks "mostly fine", I think all the values should be checked because the end of the first interval is also important and in total they are not many.